js_modules/dagster-ui/packages/ui-core/src/pipelines/JobTabs.tsx

```diff
@@ -0,0 +1,117 @@
+import {Tab, Tabs, Tooltip} from '@dagster-io/ui-components';
+import * as React from 'react';
+
+import {PermissionResult, PermissionsState, permissionResultForKey} from '../app/Permissions';
+import {TabLink} from '../ui/TabLink';
+import {RepoAddress} from '../workspace/types';
+import {workspacePathFromAddress} from '../workspace/workspacePath';
+
+import {ExplorerPath, explorerPathToString} from './PipelinePathUtils';
+
+export const DEFAULT_JOB_TAB_ORDER = ['overview', 'playground', 'runs', 'partitions'];
+
+interface Props {
+  repoAddress: RepoAddress;
+  isJob: boolean;
+  explorerPath: ExplorerPath;
+  matchingTab?: string;
+  permissions: PermissionsState;
+  tabs: JobTabConfig[];
+}
+
+export const JobTabs = (props: Props) => {
+  const {repoAddress, isJob, explorerPath, matchingTab = '', permissions, tabs} = props;
+
+  const explorerPathForTab = explorerPathToString({
+    ...explorerPath,
+    opNames: [],
+  });
+
+  const selectedTab = React.useMemo(() => {
+    return (
+      tabs.find((tab) => tab.pathComponent === matchingTab) ||
+      tabs.find((tab) => tab.pathComponent === '')
+    );
+  }, [matchingTab, tabs]);
+
+  return (
+    <Tabs size="large" selectedTabId={selectedTab!.id}>
+      {tabs
+        .filter((tab) => !tab.isHidden)
+        .map((tab) => {
+          const {id, title: text, getPermissionsResult} = tab;
+          const permissionsResult = getPermissionsResult ? getPermissionsResult(permissions) : null;
+          const disabled = !!(permissionsResult && !permissionsResult.enabled);
+          const title =
+            permissionsResult && disabled ? (
+              <Tooltip content={permissionsResult.disabledReason} placement="top">
+                {text}
+              </Tooltip>
+            ) : (
+              text
+            );
+
+          const href = workspacePathFromAddress(
+            repoAddress,
+            `/${isJob ? 'jobs' : 'pipelines'}/${explorerPathForTab}${tab.pathComponent}`,
+          );
+
+          if (disabled) {
+            return <Tab disabled key={id} id={id} title={title} />;
+          }
+
+          return <TabLink key={id} id={id} title={title} disabled={disabled} to={href} />;
+        })}
+    </Tabs>
+  );
+};
+
+export type JobTabConfigInput = {
+  hasLaunchpad: boolean;
+  hasPartitionSet: boolean;
+};
+
+export interface JobTabConfig {
+  id: string;
+  title: string;
+  pathComponent: string;
+  getPermissionsResult?: (permissionsState: PermissionsState) => PermissionResult;
+  isHidden?: boolean;
+}
+
+export const buildJobTabMap = (input: JobTabConfigInput): Record<string, JobTabConfig> => {
```

Comment:
It might be worth adding a summary comment in here about how this is used / overridden in cloud so it's easier to take into account when we're updating this.